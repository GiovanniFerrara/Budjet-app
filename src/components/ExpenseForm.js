import React, { Component } from 'react';
import moment from 'moment';
import {SingleDatePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css'

const now = moment();
console.log(now.format("Do - MMM - YYYY"))

class ExpenseForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
        description: props.expense?  props.expense.description : "",
        createdAt: props.expense?  moment(props.expense.createdAt) : moment(),
        amount: props.expense?  (props.expense.amount/100).toString() : 0,
        note: props.expense? props.expense.note : "",
        calendarFocus: false,
        error: "",
    }
  }
  handleSubmit = (e) => {
    e.preventDefault();
    if(!this.state.description || !this.state.amount ){
      this.setState(()=>({
        error : "Please enter description and amount"
      }))
    } else {
      this.setState(()=>({
        error : ""
      }))
      this.props.onSubmit({
        description: this.state.description,
        createdAt: this.state.createdAt.valueOf(),
        amount: parseFloat(this.state.amount,10)*100,
        note: this.state.note,
      })
    }
  }
  onDateChange = (createdAt)=>{
    if(createdAt){
      this.setState(()=>({createdAt}))
    }
  }
  onFocusChange = ({focused}) => {
    this.setState(()=> ({calendarFocus: focused}))
  }
  handleInputChange = (e)=>{
    const input = e.target.name;
    const value = e.target.value;
    this.setState(()=>({
      [input]: value,
    }))
  }
  render() {
    return (
      <div>
        {this.state.error && <h4> {this.state.error} </h4>}
        <form onSubmit={this.handleSubmit}>
          <input type="text"
            placeholder="description"
            name="description"
            autoFocus
            onChange={this.handleInputChange}
            noValidate = {true}
            value = {this.state.description}
            
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocus} // PropTypes.bool
            onFocusChange={this.onFocusChange}
            id="your_unique_id" // PropTypes.string.isRequired,
            numberOfMonths={2}
            isOutsideRange = {()=> false}
          />
          <input type="number"
            name="amount"
            placeholder="Amount"
            onChange={this.handleInputChange}
            value = {this.state.amount}
          />
          <textarea type="text"
            name="note"
            placeholder="Notes"
            onChange={this.handleInputChange}
            value = {this.state.note}
          />
          <button> Add Expense </button>
        </form>
      </div>
    );
  }
}

export default ExpenseForm;
