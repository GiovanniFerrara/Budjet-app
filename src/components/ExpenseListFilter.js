import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount } from '../actions/filters';

// text: '',
//   sortBy: 'date',
//   startDate: undefined,
//   endDate: undefined

const ExpenseListFilter = ({ filters, dispatch }) => {

  return (
    <div>
      <input type="text" value={filters.text} onChange={(e) => {
        dispatch(setTextFilter(e.target.value))
      }} />
      <select value={filters.sortBy} onChange={(e) => {
        e.target.value === "amount" && dispatch(sortByAmount());
        e.target.value === "date" && dispatch(sortByDate());
      }}>
        <option value="date">Date</option>
        <option value="amount"> Amount</option>
      </select>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseListFilter);
