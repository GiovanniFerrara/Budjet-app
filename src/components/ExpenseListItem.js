import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

export const ExpenseListItem = ({ description, amount, id, createdAt, dispatch }) => {
  return (
    <div>
      <h3><Link to={"edit/" + id}>{description}</Link></h3>
      <p> {amount / 100} -  {moment(createdAt).format('DD-MMMM-YYYY')} </p>
    </div>
  );
}


export default ExpenseListItem;
