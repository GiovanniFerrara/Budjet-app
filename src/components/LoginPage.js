import React, { Component } from 'react';
import authProvider from '../firebase/firebase';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth'

export class LoginPage extends Component {
  render() {
    return (
      <div>
        <h1>Login</h1>
        <button onClick={this.props.login} >Login</button>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  login: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)