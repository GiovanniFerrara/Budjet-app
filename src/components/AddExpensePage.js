import React, { Component } from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { startAddExpense } from '../actions/expenses'

export class AddExpensePage extends Component {

  onSubmit = (expense) => {
    this.props.startAddExpense(expense)
    this.props.history.push("/dashboard")
  }
  render() {
    return (
      <div>
        This is from my add expense component
    <ExpenseForm onSubmit={this.onSubmit} />
      </div>
    )
  }
}


const mapDispatchToProps = (dispatch) => ({
  startAddExpense: (expense) => {
    dispatch(startAddExpense(expense))
  }
}
)

export default connect(undefined, mapDispatchToProps)(AddExpensePage)
