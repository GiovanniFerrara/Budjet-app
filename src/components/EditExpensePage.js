import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses'

const EditExpensePage = ({dispatch,history,expense,match}) => (
  <div>
    This is from my edit expense component
    <ExpenseForm 
      onSubmit ={ (newExpense)=> {
        console.log(newExpense)
        dispatch(editExpense(expense.id,newExpense));
        history.push("/")
      }}
      expense = {expense}
    />
    
    <button onClick={()=>{
      dispatch(removeExpense({id: expense.id}))
      history.push("/")
    }
      }> Remove </button>
  </div>
);
const mapStateToProps = (state, props)=>({
  expense: state.expenses.find((expense)=>{
    return expense.id = props.match.params.id
  })
})
export default connect(mapStateToProps)(EditExpensePage)
