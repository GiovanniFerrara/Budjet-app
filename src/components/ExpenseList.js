import React, { Component } from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem'
import getVisibleExpenses from '../selectors/expenses';
import { startFetchExpenses } from '../actions/expenses'

export class ExpenseList extends Component {
  componentDidMount() {
    this.props.fetchExpenses()
  }
  render() {
    return (
      <div>
        <h1>Expense List</h1>
        {this.props.expenses.length === 0 ?
          <p>Loading...</p> :
          this.props.expenses.map((item) => (
            <ExpenseListItem {...item} key={item.id} />
          ))}
      </div>
    )
  }
}
const mapDispatchToProps = (dispatch) => ({
  fetchExpenses: () => {
    dispatch(startFetchExpenses())
  }
})

const mapStateToProps = (state) => {
  return { expenses: getVisibleExpenses(state.expenses, state.filters) }
};

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseList);
