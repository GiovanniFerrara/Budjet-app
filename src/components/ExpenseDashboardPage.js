import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseList />
    <ExpenseListFilter />
    <button onClick={() => {
      console.log(process.env.FIREBASE_API_KEY)
    }}>click</button>
  </div>
);

export default ExpenseDashboardPage;
