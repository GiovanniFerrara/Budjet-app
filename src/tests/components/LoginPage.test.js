import { LoginPage } from '../../components/LoginPage'
import { shallow } from 'enzyme';
import React from 'react';

test('should render the login page', () => {
  const wrapper = shallow(<LoginPage />)
  expect(wrapper).toMatchSnapshot()
})


test('should call login on button click', () => {
  const loginSpy = jest.fn();
  const wrapper = shallow(<LoginPage login={loginSpy} />)
  wrapper.find('button').simulate('click')
  expect(loginSpy).toHaveBeenCalled()
})