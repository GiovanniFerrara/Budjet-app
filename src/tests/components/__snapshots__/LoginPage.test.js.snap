// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the login page 1`] = `
<div>
  <h1>
    Login
  </h1>
  <button>
    Login
  </button>
</div>
`;
